From eefcdc53635aea0fc25d95ab7f1d38113b6799b3 Mon Sep 17 00:00:00 2001
From: eternalwrt-bot <zj18139624826@gmail.com>
Date: Sat, 27 Jun 2026 11:28:53 +0800
Subject: [PATCH] luci-mod-system: add modal overlay dialog to reboot

---
 .../resources/view/system/reboot.js           | 21 ++++++++++++++++++-
 1 file changed, 20 insertions(+), 1 deletion(-)

diff --git a/modules/luci-mod-system/htdocs/luci-static/resources/view/system/reboot.js b/modules/luci-mod-system/htdocs/luci-static/resources/view/system/reboot.js
index 92e1dd4..c106f78 100644
--- a/modules/luci-mod-system/htdocs/luci-static/resources/view/system/reboot.js
+++ b/modules/luci-mod-system/htdocs/luci-static/resources/view/system/reboot.js
@@ -30,7 +30,26 @@ return view.extend({
 		body.appendChild(E('hr'));
 		body.appendChild(E('button', {
 			'class': 'cbi-button cbi-button-action important',
-			'click': ui.createHandlerFn(this, 'handleReboot')
+			'click': function () {
+				ui.showModal(_('确认重启'), [
+					E('p', {}, _('你确认要重启系统？')),
+					E('button', {
+						'class': 'cbi-button cbi-button-action important',
+						'style': 'margin-left: 0px; background: red!important; border-color: red!important',
+						'click': function () {
+							ui.hideModal();
+							this.handleReboot();
+						}.bind(this)
+					}, _('确认')),
+					E('button', {
+						'class': 'btn cbi-button cbi-button-apply',
+						'style': 'margin-left: 20px',
+						'click': function () {
+							ui.hideModal();
+						}
+					}, _('Cancel'))
+				]);
+			}.bind(this)
 		}, _('Perform reboot')));
 
 		return body;
-- 
2.42.0
